### ISSUE TITLE
Missing packages when running nixos tests

### ISSUE THREAD

**Issue by jan-kouba** [NixOS/nixpkgs#264577]
### Describe the bug
Packages are missing when running nixos tests.

### Steps To Reproduce

Flake:
```
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
  };

  outputs = { self, nixpkgs }: 
  let 
    system = "x86_64-linux";
    nixos-lib = import (nixpkgs + "/nixos/lib") {};
    hostPkgs = nixpkgs.legacyPackages.${system};
  in {
    checks.${system}.nixosTest = nixos-lib.runTest {
      name = "simple test";

      nodes.machine = { pkgs, ... }: {
        environment.systemPackages = [ 
          pkgs.gitAndTools.git
        ];
      };

      hostPkgs = hostPkgs;

      testScript = ''
        machine.start(allow_reboot = True)

        machine.wait_for_unit("multi-user.target")
        machine.wait_until_succeeds("pgrep -f 'agetty.*tty1'")
        machine.screenshot("postboot")

        with subtest("git"):
          machine.succeed('mkdir -p /tmp/git-test')
          machine.succeed('cd /tmp/git-test')
          machine.succeed('git init /tmp/git-test')
      '';
    };
  };
}
```

Running `nix build .#checks.x86_64-linux.nixosTest` fails with the error:
```console
warning: Git tree '/tmp/nixos-test' is dirty
error: attribute 'gitAndTools' missing

 at /nix/store/5m9801ag8r5w13fl1jdvsj6pr66znrbr-source/flake.nix:17:11:

     16|         environment.systemPackages = [
     17|           pkgs.gitAndTools.git
       |           ^
     18|         ];
(use '--show-trace' to show detailed location information)
```

If I change the line `pkgs.gitAndTools.git` to `hostPkgs.gitAndTools.git` or to `pkgs.git`, the build succeeds.

The same problem exists also e.g. for the `pkgs.kate` package.

### Expected behavior
`runTest` should not remove packages from `hostPkgs`.

### Additional context
I dont know, if it is an expected behaviour or some bug, but I found nothing about this behaviour in NixOS manual.

### Notify maintainers

@roberth 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.9, NixOS, 23.05 (Stoat), 23.05.20231027.1a3c95e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by jan-kouba** [comment]
Hi @kirillrdy, thanks for quick response!

OK, I understand now. I fixed the issue by not using the aliases (`pkgs.gitAndTools.git` -> `pkgs.git`, `pkgs.kate` -> `pkgs.plasma5Packages.kate`).

It would be helpful, if there is some note about this (aliases not being included) in the NixOS manual. 
Also a suggestion how to find the non-aliased package path would help. I just used `nix search nixpkgs <package-name>` and just guessed what is the non aliased path.
